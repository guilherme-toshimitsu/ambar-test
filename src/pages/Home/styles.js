const styles = {
  buttonsContainer: {
    marginTop: 50,
    marginBottom: 50,
    display: 'flex',
    justifyContent: 'center',
  },
  button: {
    marginRight: 10,
    marginLeft: 10,
  },
};

export default styles;
