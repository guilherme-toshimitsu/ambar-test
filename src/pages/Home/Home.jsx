import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '@material-ui/core';

import { useDispatch, useSelector } from 'react-redux';
import { actions } from '@store/temperature';
import WeatherViewer from '@components/WeatherViewer';
import { makeStyles } from '@material-ui/styles';
import styles from './styles';

const useStyles = makeStyles(styles, { name: 'Home' });

const Home = ({ history }) => {
  const classes = useStyles();
  const temperature = useSelector(state => state.temperature);
  const dispatch = useDispatch();

  return (
    <>
      <div className={classes.buttonsContainer}>
        <Button
          className={classes.button}
          disabled={temperature.isLoading}
          onClick={() => dispatch(actions.getTemperature({ lat: 40.73061, lon: -73.935242 }))}
          color="primary"
          variant="contained"
        >
          NY
        </Button>
        <Button
          className={classes.button}
          disabled={temperature.isLoading}
          onClick={() => dispatch(actions.getTemperature({ lat: -23.54, lon: -46.63 }))}
          color="primary"
          variant="contained"
        >
          SP
        </Button>
        <Button
          className={classes.button}
          disabled={temperature.isLoading}
          onClick={() => dispatch(actions.getTemperature({ lat: 35.652832, lon: 139.839478 }))}
          color="primary"
          variant="contained"
        >
          TOKYO
        </Button>
      </div>
      <WeatherViewer
        main={temperature.main}
        weather={temperature.weather}
        isLoading={temperature.isLoading}
        name={temperature.name}
        history={history}
      />
    </>
  );
};

Home.propTypes = {
  history: PropTypes.object,
};

export default Home;
