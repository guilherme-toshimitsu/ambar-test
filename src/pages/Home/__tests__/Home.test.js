import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Home from '../';
import { Provider } from 'react-redux';
import configureStore from '@store';

describe('<Home> Home Page Container', () => {
  it('renders', () => {
    const Store = configureStore();
    const container = render(
      <Provider store={Store}>
        <Home />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
