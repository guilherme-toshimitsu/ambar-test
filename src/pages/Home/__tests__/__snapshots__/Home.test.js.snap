// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home> Home Page Container renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Home-buttonsContainer-1"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Home-button-2 MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            NY
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Home-button-2 MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            SP
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Home-button-2 MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            TOKYO
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation3 WeatherViewer-paper-34 MuiPaper-rounded"
      >
        <div
          class="WeatherViewer-cardContainer-36"
        >
          Clique em um Botão escolher a cidade
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Home-buttonsContainer-1"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained Home-button-2 MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          NY
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained Home-button-2 MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          SP
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained Home-button-2 MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          TOKYO
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation3 WeatherViewer-paper-34 MuiPaper-rounded"
    >
      <div
        class="WeatherViewer-cardContainer-36"
      >
        Clique em um Botão escolher a cidade
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
